### ISSUE TITLE
--env-file does not expose variables to detect

### ISSUE THREAD

**Issue by nebhale** [buildpack/pack#83]
In the latest revision of the spec, [environment variables are exposed during detection](https://github.com/buildpack/spec/blob/master/buildpack.md#provided-by-the-platform).  At the moment, these environment variable are only exposed during the build phase.  They should be exposed during the detect phase as well.

**Comment by sclevine** [comment]
Looks good:
```
liberty:pack pivotal$ ./pack-676-osx build myapp --builder packs/samples:0.0.1-rc.490 --buildpack ./test-bp/ --env-file <(echo 'TESTME=pass')
Defaulting app directory to current working directory /Users/pivotal/workspace/cncf-buildpacks/pack (use --path to override)
Using user-provided builder image packs/samples:0.0.1-rc.490
Pulling builder image packs/samples:0.0.1-rc.490 (use --no-pull flag to skip this step)
Selected run image packs/run:v3alpha2 from stack io.buildpacks.stacks.bionic
Pulling run image packs/run:v3alpha2 (use --no-pull flag to skip this step)
Using cache volume pack-cache-7bae996a0e238a80f14b74a61e3178bf
===> DETECTING
Using manually-provided group
[detector] 2019/01/04 16:01:59 Trying group of 1...
[detector] 2019/01/04 16:01:59 ======== Output:  ========
[detector]
[detector] 2019/01/04 16:01:59 ======== Results ========
[detector] 2019/01/04 16:01:59 : pass
```

**Comment by ekcasey** [comment]
released in `pack` `v0.0.9`
